analyzers/tests/SonarAnalyzer.UnitTest/MetadataReferences/NuGetMetadataFactory.cs

```diff
@@ -66,67 +64,36 @@ public static partial class NuGetMetadataFactory
-                if (!Directory.Exists(packageDir))
-                {
-                    throw new ApplicationException($"Test setup error: folder for downloaded package does not exist. Folder: {packageDir}");
-                }
-            }
-
-            return Directory.GetFiles(packageDir, "*.dll", SearchOption.AllDirectories)
-               .Select(x => (MetadataReference)MetadataReference.CreateFromFile(x))
-               .ToImmutableArray();
-        }
-
         /// <param name="allowedDirectories">List of allowed directories sorted by preference to search for DLL files.</param>
         private static IEnumerable<MetadataReference> Create(Package package, string[] allowedDirectories)
         {
-            Console.WriteLine();
-            Console.WriteLine($"Package: {package.Id}, {package.Version}");
             package.EnsurePackageIsInstalled();
-
-            var packageDirectory = package.PackageDirectory();
-            if (!Directory.Exists(packageDirectory))
+            var dllsPerDirectory = new Dictionary<string, IEnumerable<string>>();
+            var packageLibDirectory = package.PackageDirectory();
+            var foldersWithDlls = new List<string> { Path.Combine(packageLibDirectory, "lib"), Path.Combine(packageLibDirectory, "ref") };
+            foreach (var folder in foldersWithDlls.Where(Directory.Exists))
             {
-                throw new ApplicationException($"Test setup error: Package directory doesn't exist: {packageDirectory}");
+                dllsPerDirectory.AddRange(DllsPerDirectory(folder).Where(x => !dllsPerDirectory.ContainsKey(x.Key)));
             }
-            var dllsPerDirectory = Directory.GetFiles(packageDirectory, "*.dll", SearchOption.AllDirectories)
-                .GroupBy(x => new FileInfo(x).Directory.Name)
-                .ToDictionary(x => x.Key.Split('+').First(), x => x.AsEnumerable());
 
             // if this throws because it doesn't find a DLL (because the lib contains "_._"), maybe you are referencing the wrong DLL (check the dependencies of the DLL you use)
             var directory = allowedDirectories.FirstOrDefault(x => dllsPerDirectory.ContainsKey(x))
-                ?? throw new InvalidOperationException($"No allowed directory with DLL files was found in {packageDirectory}. " +
-                                                        "Add new target framework to SortedAllowedDirectories or set targetFramework argument explicitly.");
-            var dlls = dllsPerDirectory[directory];
-            foreach (string filePath in dlls)
+                            ?? throw new InvalidOperationException($"No allowed directory with DLL files was found in {packageLibDirectory}. " +
+                                                                   "Add new target framework to SortedAllowedDirectories or set targetFramework argument explicitly.");
+            var dlls = dllsPerDirectory[directory].ToList();
+            foreach (var filePath in dlls)
```

Comment:
We don't need the list, we'll just iterate it once. Improving `foreach` variable name can remove the `dlls`.
```suggestion
            foreach (var dllPath in dllsPerDirectory[directory])
```